query/src/test/scala/filodb/query/PlanValidationSpec.scala

```diff
@@ -1,24 +1,66 @@
 package filodb.query
 
 import org.scalatest.matchers.should.Matchers
-
 import filodb.query.exec.ExecPlan
 
+import scala.collection.mutable
+
 trait PlanValidationSpec extends Matchers {
 
+  private def getIndent(line: String): Int = {
+    line.prefixLength(c => c == '-')
+  }
+
+  private def removeNoise(planString: String): String = {
+    planString
+      .replaceAll("testProbe-.*]", "testActor]")
+      .replaceAll("InProcessPlanDispatcher.*\\)", "InProcessPlanDispatcher")
+  }
+
+  /**
+   * Prints the same tree for all logically-identical plans.
+   */
+  private def sortTree(planString: String): Unit = {
```

Comment:
I got tired of tests failing due to children randomly getting swapped in `ExecPlan::printTree`. This is disabled by default, but you can use this to consistently sort the `String` trees before they're passed to `shouldEqual`.